debezium-embedded/src/main/java/io/debezium/connector/simple/SimpleSourceConnector.java

```diff
@@ -166,19 +194,21 @@ public void start(Map<String, String> props) {
                         }
                         SourceRecord record = new SourceRecord(partition, offset, topic, 1, keySchema, key, valueSchema, value);
                         records.add(record);
+                        postRecord(recordNum, id, record);
                     }
+                    postBatch(batch);
                 }
             }
         }
 
         @Override
         public List<SourceRecord> poll() throws InterruptedException {
-            if (records.isEmpty()) {
-                // block forever, as this thread will be interrupted if/when the task is stopped ...
-                new CountDownLatch(1).await();
+            while (running.get() && records.isEmpty()) {
```

Comment:
perhaps handling interruptions could also be done here - i.e. adding a `&& !Thread.interrupted()`